src/DockSettings.vala

```diff
@@ -0,0 +1,23 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0
+ * SPDX-FileCopyrightText: 2025 elementary, Inc. (https://elementary.io)
+ */
+
+public class Dock.DockSettings : GLib.Object {
```

Comment:
I don't love this. We've had folks make settings classes in the past and we always end up removing it because it was a bad idea and GLib.Settings already does everything we need